demoapp/testimony_summary.py

```diff
@@ -0,0 +1,204 @@
+
+bills_to_select = {
+    '#H3121': 'An Act relative to the open meeting law',
+    '#S2064': 'An Act extending the public records law to the Governor and the Legislature',
+    '#H711': 'An Act providing a local option for ranked choice voting in municipal elections',
+    '#S1979': 'An Act establishing a jail and prison construction moratorium',
+    '#H489': 'An Act providing affordable and accessible high-quality early education and care to promote child development and well-being and support the economy in the Commonwealth',
+    '#S2014': 'An Act relative to collective bargaining rights for legislative employees',
+    '#S301': 'An Act providing affordable and accessible high quality early education and care to promote child development and well-being and support the economy in the Commonwealth',
+    '#H3069': 'An Act relative to collective bargaining rights for legislative employees',
+    '#S433': 'An Act providing a local option for ranked choice voting in municipal elections',
+    '#H400': 'An Act relative to vehicle recalls',
+    '#H538': 'An Act to Improve access, opportunity, and capacity in Massachusetts vocational-technical education',
+    '#S257': 'An Act to end discriminatory outcomes in vocational school admissions'
+}
+
+def find_bills(bill_number, df = mgl_df):
+    """input:
+    args: bill_number: (str), Use the number of the bill to find its title and content
+    """
+    bill = df[df['BillNumber'] == bill_number]['DocumentText']
+
+    try:
+         # Locate the index of the bill
+        idx = bill.index.tolist()[0]
+        # Locate the content and bill title of bill based on idx
+        content = df['DocumentText'].iloc[idx]
+        bill_title = df['Title'].iloc[idx]
+        bill_number = df['BillNumber'].iloc[idx]
+        # laws
+        # law = df['combined_MGL'].iloc[idx]
+
+        return content, bill_title, bill_number
+    
+    except Exception as e:
+        content = "blank"
+        print(e)
+
+def generate_categories(text):
+
+    category_prompt = """According to this list of category {category}.
+
+        classify this bill {context} into a closest relevant category.
+
+        Do not output a category outside from the list
+    """
+
+    prompt = PromptTemplate(template=category_prompt, input_variables=["context", "category"])
+    llm = LLMChain(llm = ChatOpenAI(openai_api_key=open_ai_key, temperature=0, model=MODEL_NAME), prompt = prompt)
+        
+    response = llm.predict(context = text, category = category_for_bill) # grab from tagging.py
+    return response
+
+def cache(func):
+    def w(arg):
+        if arg in SUMMARY_CACHE: return SUMMARY_CACHE[arg]
+        else: 
+            out = func(arg)
+            SUMMARY_CACHE[arg] = out
+            with open('generated_summary.json', 'w') as json_file: json.dump(SUMMARY_CACHE, json_file)
+            return out
+    return w
+
+@cache
+def _generate_summary(bill_id, df = mgl_df):
```

Comment:
Is this meant to be the same as what's in the demo_app script? If so, can we import from there instead of redefining? Or is this new logic?

Reply:
Looks like SQLite cache is a better idea, as it caches all the calls instead of specific ones that we add the decorator for.